Cherry-pick Qt 5.14 build fixes, and new windows build env and deps from master branch to v1.17-dev branch

anthonybilinski opened the pull request:
Cherry-pick Qt 5.14 build fixes, and new windows build env and deps from `master` to `v1.17-dev`.

- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/5858)
<!-- Reviewable:end -->

Chiitoo committed:
refactor(Qt): use 'QLatin1String' instead of 'QLatin1Literal'

This is an undocumented Qt 2/3/4 remnant, deprecated in Qt 5.14.

- https://code.qt.io/cgit/qt/qtbase.git/commit/?h=5.14&id=3e75c296
- https://code.qt.io/cgit/qt/qtbase.git/commit/?h=5.14&id=45373c19

Chiitoo committed:
refactor(Qt): use 'asprintf' instead of 'sprintf'

Deprecated since Qt 5.14.

- https://code.qt.io/cgit/qt/qtbase.git/commit/?h=5.14&id=f8f592d5

nurupo committed:
feat: add Fcitx and Uim support to AppImage

anthonybilinski committed:
chore(tools): allow merging PR to whatever branch is checked out

Useful for either dev testing or merging PRs to dev branches. Also stop
updating local repo state beyond merging the PR.

nurupo committed:
feat(build): use Debian Buster for Windows cross-compilation

nurupo committed:
chore(windows): update OpenSSL to 1.1.1d

nurupo committed:
chore(windows): update Qt to 5.12.5

nurupo committed:
chore(windows): update FFmpeg to 4.2.1

nurupo committed:
chore(windows): update VPX to 1.8.1

nurupo committed:
fix(build): fix SQLCipher build issue with OpenSSL 1.1.1 on Windows

nurupo committed:
fix(build): fix OpenAL Soft failing to build with newer CMake

Chiitoo committed:
refactor(Qt): use 'QLatin1String' instead of 'QLatin1Literal' part 2

This is required when building against Qt 5.14.

Chiitoo committed:
refactor(Qt): use 'storeRelaxed()' and 'loadRelaxed()'

This fixes build with Qt 5.14 by replacing the deprecated
QAtomic::load() and store() functions with the new ones.

- https://code.qt.io/cgit/qt/qtbase.git/commit/?h=5.14&id=79bdc7cf

sudden6 reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r1, 4 of 4 files at r2, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5858#-:-LqL7lrf-EKEzZkcGLqW:b-jbsf8z)* status: :shipit: complete! 1 of 1 LGTMs obtained



<!-- Sent from Reviewable.io -->

anthonybilinski committed:
chore(travis): make travis happy by having some output - but not too much

Fix #5515

brew install might take a long time to build Qt or ffmpeg. Travis kills us if
we don't output for 10 minutes. Travis also kills us if we output too much, so
verbose isn't an option. So just output some dots...

anthonybilinski committed:
fix(core): avoid logging which bootstrap node is being connected to

Errors are parsed and printed, but which specific node is being connected to
isn't very relevant to any errors that would occur, and the nodes list is
already updated to prune offline nodes based on nodes.tox.chat periodically.
This provides some extra privacy about which connections are being made,
even though the bootstrap nodes are already public.

anthonybilinski committed:
fix(ci): upgrade travis to xcode9.3 to get macos 10.13 with brew support

Brew now only officially supports Mac OS 10.13-10.15, which means our
previously used 10.12 had fewer binary packages available, meaning we had to
compile more ourselves, causing CI to timeout.
https://docs.travis-ci.com/user/reference/osx#macos-version
https://github.com/Homebrew/brew/pull/6500

Fix #5515

anthonybilinski committed:
fix(osx): create /usr/local/sbin directory for brew package installation

ffmpeg dependency unbound attempts to symlink to /usr/local/sbin, failing if
the directory does not exist. /usr/local/sbin does not exist by default on
10.13 High Sierra, so prompt to create it if it doesn't exit, and create it for
travis CI automatically.
